torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -160,9 +160,11 @@ def inner(*flat_args):
             flat_f_args = pytree.tree_map(_to_fun, flat_args)
             flat_f_outs = f(*flat_f_args)
             # We didn't do any tracing, so we don't need to process the
-            # unbacked symbols, they will just disappear into the ether
+            # unbacked symbols, they will just disappear into the ether.
+            # Also, prevent memoization from applying.
             if (fake_mode := detect_fake_mode()) and (shape_env := fake_mode.shape_env):
                 shape_env.pending_fresh_unbacked_symbols.clear()
+                fake_mode.epoch += 1
```

Comment:
Not super familiar here but comment says
```
# Also, prevent memoization from applying.
```

Here, we are also invalidating old caches. Is that intended ? and it's also making subsequent runs pick up our cache from this run

Reply:
Yes. The point is when we retrace, we need to ensure we generate accurate `unbacked_bindings`. To do that, we need to actually generate fresh symbols, so that we know that something is a binding site.